_shared/project/pyproject.toml

```diff
@@ -21,6 +21,90 @@ filterwarnings = [
     {% endif %}
 ]
 
+{% if cookiecutter.get("linter") == "ruff" %}
+[tool.ruff]
+target-version = "py311"
+line-length = 88
+exclude = [
+    "tests/bdd/steps/_compiled_feature_steps.py",
+]
+
+[tool.ruff.lint]
+select = [
+  "E", "W", #  https://docs.astral.sh/ruff/rules/#pycodestyle-e-w
+  "D", # https://docs.astral.sh/ruff/rules/#pydocstyle-d
+  "ARG", # https://docs.astral.sh/ruff/rules/#flake8-unused-arguments-arg
+  "BLE001", # https://docs.astral.sh/ruff/rules/blind-except/
+  "R", "PLR", # https://docs.astral.sh/ruff/rules/#refactor-r
+  "C", "PLC", # https://docs.astral.sh/ruff/rules/#convention-c
+  "SLF", # flake-8-self
+  "N", # https://docs.astral.sh/ruff/rules/#pep8-naming-n
+  "F", # https://docs.astral.sh/ruff/rules/unused-import/
+
+  "RUF100", # unused-noqa
+]
+
+ignore = [
+    # Missing docstrings.
+    "D100","D101","D102","D103","D104","D105","D106","D107",
+
+    # "No blank lines allowed after function docstring" conflicts with the
+    # Black code formatter which insists on inserting blank lines after
+    # function docstrings.
+    "D202",
+
+    # "1 blank line required before class docstring" conflicts with another
+    # pydocstyle rule D211 "No blank lines allowed before class docstring".
+    "D203",
+
+    # "Multi-line docstring summary should start at the first line"
+    # and "Multi-line docstring summary should start at the second line".
+    # These two rules conflict with each other so you have to disable one of them.
+    # How about we disable them both? PEP 257 says either approach is okay:
+    #
+    # > The summary line may be on the same line as the opening quotes or on
+    # > the next line.
+    # >
+    # > https://peps.python.org/pep-0257/#multi-line-docstrings
+    "D212",
+    "D213",
+
+    # We use Black to format our code automatically, so we don't need PyLint to
+    # check formatting for us.
+    "E501", # line-too-long
+
+    "PLR2004", # Magic values, we mostly get it on HTTP status codes
+
+    # Disabled during the pylint migration, ideally we'll enable this after we are settled in ruff
+    "RET504",
```

Comment:
Having these here allows to remove them from the template and apply to all projects.